gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java

```diff
@@ -277,28 +295,45 @@ private void addMapPin(MapPin mapPin) {
   }
 
   private void addMapPolyline(MapPolygon mapPolygon) {
-    for (ImmutableSet<Point> vertices : mapPolygon.getVertices()) {
-      PolylineOptions options = new PolylineOptions();
-
-      // Read-only
-      options.clickable(false);
-
-      // Add vertices to PolylineOptions
-      stream(vertices).map(GoogleMapsMapAdapter::toLatLng).forEach(options::add);
+    PolylineOptions options = new PolylineOptions();
+    // Read-only
+    options.clickable(false);
+    // Add vertices to PolylineOptions
+    ImmutableList<LatLng> vertices = stream(mapPolygon.getVertices())
+        .map(GoogleMapsMapAdapter::toLatLng)
+        .collect(toImmutableList());
+    stream(vertices).forEach(options::add);
+    // Add to map
+    Polyline polyline = map.addPolyline(options);
+    polyline.setTag(mapPolygon);
+    // Style polyline
+    if (!isPolygonCompleted(mapPolygon.getVertices())) {
+      BitmapDescriptor customCap = bitmapDescriptorFromVector();
+      polyline.setStartCap(new CustomCap(customCap));
+      polyline.setEndCap(new CustomCap(customCap));
+    }
+    polyline.setWidth(getPolylineStrokeWidth());
+    polyline.setColor(parseColor(mapPolygon.getStyle().getColor()));
+    polyline.setJointType(JointType.ROUND);
 
-      // Add to map
-      Polyline polyline = map.addPolyline(options);
-      polyline.setTag(mapPolygon);
+    polylines.add(polyline);
+    polygons.put(mapPolygon, vertices);
+  }
 
-      // Style polyline
-      polyline.setStartCap(new RoundCap());
-      polyline.setEndCap(new RoundCap());
-      polyline.setWidth(getPolylineStrokeWidth());
-      polyline.setColor(parseColor(mapPolygon.getStyle().getColor()));
-      polyline.setJointType(JointType.ROUND);
+  private boolean isPolygonCompleted(List<Point> vertices) {
+    return vertices.size() != stream(vertices).distinct().count();
```

Comment:
Not sure this is correct; "In geometry, a polygon (/ˈpɒlɪɡɒn/) is a plane figure that is described by a finite number of straight line segments connected to form a closed polygonal chain (or polygonal circuit). "
https://en.wikipedia.org/wiki/Polygon

In that case, we should be looking if the first and last vertices are equivalent, not if all vertices are distinct as you're doing here. Also it implies the method should be called "isClosedPolygonalChain", with a comment that clarifies that it's checking if "the first and last vertices match such that they form a closed polygonal chain."

Reply:
updated.